lib/vnet/local_app_provider.go

```diff
@@ -0,0 +1,216 @@
+package vnet
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+
+	apiclient "github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+	vnetv1 "github.com/gravitational/teleport/gen/proto/go/teleport/lib/vnet/v1"
+	"github.com/gravitational/teleport/lib/auth/authclient"
+)
+
+// ClientApplication is the common interface implemented by each VNet client
+// application: Connect and tsh. It provides methods to list user profiles, get
+// cluster clients, issue app certificates, and report metrics and errors.
+type ClientApplication interface {
```

Comment:
`ClientApplication` is one of those names that at first seems weird, but makes sense once you understand what's happening. Though I have to say that the `ClientApplicationService` proto service seems to have a slightly better comment. It explicitly mentions the admin process, so I think it makes it easier to understand why this thing is named "client application".

Reply:
updated the comment, hopefully it's more clear